### PR TITLE
snap refactor 3: use plain file for CF_LOCK and little tidy up

### PR DESCRIPTION
**Pull request by choleraehyq**
@zhangjinpeng1987 @siddontang @hhkbp2 PTAL

### CODE REVIEW

File: src/raftstore/store/snap.rs [hunk 1]
```diff
@@ -672,6 +694,61 @@ impl Snap {
     }
 }
 
+pub fn build_plain_cf_file<E: BytesEncoder>(encoder: &mut E,
```
Comment (siddontang) [inline src/raftstore/store/snap.rs]: do we have test to cover plain now?

Comment (choleraehyq) [inline src/raftstore/store/snap.rs]: Yes. See https://coveralls.io/builds/12470885/source?filename=src%2Fraftstore%2Fstore%2Fsnap.rs

File: src/raftstore/store/snap.rs [hunk 2]
```diff
@@ -789,10 +866,21 @@ impl Snapshot for Snap {
...
+                try!(apply_plain_cf_file(&mut file, &options, cf_handle));
+            } else {
+                // we move instead of copy file when ingest_external_file_cf
+                // so method delete will not sub size_track
+                if file_exists(&cf_file.path) {
+                    let mut size_track = self.size_track.wl();
+                    *size_track = size_track.saturating_sub(cf_file.size);
+                }
+                let mut ingest_opt = IngestExternalFileOptions::new();
+                ingest_opt.move_files(true);
```
Comment (siddontang) [inline src/raftstore/store/snap.rs]: seem it is unsafe to move file here. 

/cc @BusyJay @hhkbp2

File: src/raftstore/store/snap.rs [hunk 3]
```diff
@@ -789,10 +866,21 @@ impl Snapshot for Snap {
...
-
-            let ingest_opt = IngestExternalFileOptions::new();
-            let path = cf_file.path.as_path().to_str().unwrap();
-            box_try!(options.db.ingest_external_file_cf(cf_handle, &ingest_opt, &[path]));
+            if plain_file_used(cf_file.cf) {
+                let mut file = box_try!(File::open(&cf_file.path));
+                try!(apply_plain_cf_file(&mut file, &options, cf_handle));
+            } else {
+                // we move instead of copy file when ingest_external_file_cf
+                // so method delete will not sub size_track
```
Comment (zhangjinpeng1987) [inline src/raftstore/store/snap.rs]: How to make size_track works?

File: src/raftstore/store/snap.rs [hunk 4]
```diff
@@ -672,6 +694,61 @@ impl Snap {
...
+        if batch_size >= options.write_batch_size {
+            box_try!(options.db.write(wb));
+            wb = WriteBatch::new();
+            batch_size = 0;
+        }
+    }
+    Ok(())
+}
+
+
```
Comment (hhkbp2) [inline src/raftstore/store/snap.rs]: Remove blank lines.

File: src/raftstore/store/snap.rs (hunk 2 above)
Comment (zhangjinpeng1987) [inline src/raftstore/store/snap.rs]: For rocksdb it is safe.

Comment (siddontang) [inline src/raftstore/store/snap.rs]: It is not for the Rocksdb, it is for re-applying snapshot of process restarts.

**Comment by siddontang** [comment]
LGTM

**Pull request merged**